op-service/txmgr/txmgr.go

```diff
@@ -114,7 +125,9 @@ type ETHBackend interface {
 // SimpleTxManager is a implementation of TxManager that performs linear fee
 // bumping of a tx until it confirms.
 type SimpleTxManager struct {
-	cfg     Config // embed the config directly
+	cfg     Config     // embed the config directly
+	cfgLock sync.Mutex // protects values that can be changed at runtime via rpc
```

Comment:
If we go with a mutex, let's better use an `RWMutex` and only `RLock` where we read the value, so everywhere except in the API setters.

Given that all accesses to the fields are simply lock-guaded read or write ops, we could use [atomics](https://pkg.go.dev/sync/atomic) instead, so read and write the fields in the config with `atomic.(Load|Store)(Uint64|Int64|Pointer)`, where you can convert `time.Duration` from/to `int64` and use an unsafe pointer for `*big.Int`. You could then encapsulate those read/write calls in the already present getters and setters, and use the getters also in the txmgr implementation where you read the values.

Alternatively you could create an `atomicConfig` struct that is an atomic version of the `Config` struct for all fields that are changeable via the API and use that `atomicConfig` inside the `SimpleTxManager` instead, which could be created from a `Config` via a `func (Config) atomic() *atomicConfig`. You could then use the `atomic.(Uint64|Int64|Pointer)` types. The nice thing about `Pointer` is that it's type-safe because it uses generics.

Reply:
This commit contains the RWMutex implementation: https://github.com/ethereum-optimism/optimism/pull/10897/commits/7b6e2e6c801ca8421a860d54cf4b8c03274460df.

This commit contains the atomics implementation: https://github.com/ethereum-optimism/optimism/pull/10897/commits/0ec063c363e4640db7715e8fa1da7b55c2be74fb.

I like the atomics option best since it allows us to lock only the individual data field we plan to read/write.